dali/operators/generic/flip.h

```diff
@@ -58,23 +58,29 @@ class Flip: public Operator<Backend> {
     return this->spec_.template GetArgument<int>("depthwise", &ws, idx);
   }
 
-  std::vector<int> GetHorizontal(const ArgumentWorkspace &ws) {
+  std::vector<int> GetHorizontal(const workspace_t<Backend> &ws) {
+    auto curr_batch_size = ws.GetInputBatchSize(0);
```

Comment:
Why not make `curr_batch_size` a function argument and ask the caller to provide it instead of querying it every time here?